solidity/contracts/LinkEx.sol

```diff
@@ -1,15 +1,34 @@
 pragma solidity 0.4.24;
 
 import "./interfaces/LinkExInterface.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
 /**
  * @title The LINK exchange contract
  */
-contract LinkEx is LinkExInterface {
+contract LinkEx is LinkExInterface, Ownable {
+  using SafeMath for uint256;
 
+  struct Rate {
+    uint256 blockNumber;
+    uint256 rate;
+  }
+
+  uint256 public constant VALID_BLOCKS = 25;
+
+  mapping(address => bool) public authorizedNodes;
+
+  mapping(address => Rate) private rates;
   uint256 private historicRate;
   uint256 private rate;
   uint256 private rateHeight;
+  address[] private oracles;
```

Comment:
Do we need this *and* authorizedNodes above?

Reply:
`authorizedNodes` is a mapping, and Solidity doesn't let you iterate over mappings.